### PR TITLE
ci(gh-actions): Update `MAINNET_RPC` and adjust requirements for self-hosted runners per job

### PR DESCRIPTION
**Pull request by Xearty**

### CODE REVIEW

**Review by Dimo99 — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ env:
   INFURA_API_KEY: '${{ secrets.INFURA_API_KEY }}'
   USER_PRIVATE_KEY: '${{ secrets.USER_PRIVATE_KEY }}'
   ETHERSCAN_API_KEY: '${{ secrets.ETHERSCAN_API_KEY }}'
-  MAINNET_RPC: https://rpc.mevblocker.io
+  MAINNET_RPC: http://192.168.8.207:8545
```
Comment (PetarKirov) [inline .github/workflows/ci.yml:16]:
```suggestion
  MAINNET_RPC: variables.MAINNET_RPC
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ env:
   INFURA_API_KEY: '${{ secrets.INFURA_API_KEY }}'
   USER_PRIVATE_KEY: '${{ secrets.USER_PRIVATE_KEY }}'
   ETHERSCAN_API_KEY: '${{ secrets.ETHERSCAN_API_KEY }}'
-  MAINNET_RPC: https://rpc.mevblocker.io
+  MAINNET_RPC: variables.MAINNET_RPC
```
Comment (PetarKirov) [inline .github/workflows/ci.yml:16]:
```suggestion
  MAINNET_RPC: variables.MAINNET_RPC
  RUST_BACKTRACE: 1
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -13,7 +13,8 @@ env:
...
+  RUST_BACKTRACE: 1
```
Comment (PetarKirov) [inline .github/workflows/ci.yml:17]:
```suggestion
  MAINNET_RPC: ${{variables.MAINNET_RPC}}
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -13,7 +13,7 @@ env:
   INFURA_API_KEY: '${{ secrets.INFURA_API_KEY }}'
   USER_PRIVATE_KEY: '${{ secrets.USER_PRIVATE_KEY }}'
   ETHERSCAN_API_KEY: '${{ secrets.ETHERSCAN_API_KEY }}'
-  MAINNET_RPC: https://rpc.mevblocker.io
+  MAINNET_RPC: ${{variables.MAINNET_RPC}}
```
Comment (PetarKirov) [inline .github/workflows/ci.yml:16]:
```suggestion
  MAINNET_RPC: '${{ variables.MAINNET_RPC }}'
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -13,7 +13,7 @@ env:
   INFURA_API_KEY: '${{ secrets.INFURA_API_KEY }}'
   USER_PRIVATE_KEY: '${{ secrets.USER_PRIVATE_KEY }}'
   ETHERSCAN_API_KEY: '${{ secrets.ETHERSCAN_API_KEY }}'
-  MAINNET_RPC: https://rpc.mevblocker.io
+  MAINNET_RPC: '${{ variables.MAINNET_RPC }}'
```
Comment (PetarKirov) [inline .github/workflows/ci.yml:16]:
```suggestion
  MAINNET_RPC: '${{ secrets.MAINNET_RPC }}'
```

**Review by PetarKirov — APPROVED** [review]

**Review by kkirkov — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -13,10 +13,10 @@ env:
   INFURA_API_KEY: '${{ secrets.INFURA_API_KEY }}'
   USER_PRIVATE_KEY: '${{ secrets.USER_PRIVATE_KEY }}'
   ETHERSCAN_API_KEY: '${{ secrets.ETHERSCAN_API_KEY }}'
-  MAINNET_RPC: https://rpc.mevblocker.io
+  MAINNET_RPC: '${{ secrets.MAINNET_RPC }}'
```
Comment (PetarKirov) [inline .github/workflows/ci.yml:16]: I have configured the GH actions secrets of the repo to point this variable to our local Erigon mainnet node.

**Pull request merged**